app/assets/javascripts/modules/media_viewer.js

```diff
@@ -47,6 +48,55 @@
         .addThumbnailsToSlider(thumbsForSlider());
     }
 
+    function _timedOut(start, time) {
+      if ((Date.now() - start) < time) {
+        return false;
+      } else {
+        return true;
+      }
+    }
+
+    function authLink(loginService, mediaObject) {
+      return jQuery('<a data-auth-link="true"></a>')
+        .prop('href', loginService['@id'])
+        .prop('target', '_blank')
+        .text(loginService.label)
+        .on('click', function(e) {
+          e.preventDefault();
+          var windowReference = window
+            .open(jQuery(this).prop('href'));
+
+            var start = Date.now();
+            var checkWindow = setInterval(function() {
+              if (!_timedOut(start, 30000) &&
+                (!windowReference || !windowReference.closed)) return;
+              clearInterval(checkWindow);
+              mediaObject[0].load();
+              mediaObject
+                .parent('[data-slider-object]')
+                .find('[data-auth-link]')
+                .hide();
+              return;
+            }, 500);
+        });
+    }
+
```

Comment:
There some odd indentation going on here? See line 69-80. Could we also have some comments to explain what's happening here?